internal/controller/api/v1alpha1/installenv_types.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2020.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type InstallEnvSpec struct {
+	// Proxy defines the proxy settings for agents and clusters that use the
+	// InstallEnv. If unset, the agents and clusters will not be configured to use a
+	// proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to 
+	// all cluster hosts. They are added to any NTP sources that were configured through
+	// other means.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// SSHAuthorizedKeys is a list of SSH public keys that will be added to all agents 
+	// for use in debugging.
+	// +optional
+	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+
+	// PullSecretRef is the reference to the secret to use when pulling images.
+	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef"`
+
+	// Agents that boot as a result of this InstallEnv should be labeled such that
```

Comment:
The [go doc convention](https://blog.golang.org/godoc) is to start the comment with the name of the thing being documented, in this case `AgentLabelSelector`. Same applies to the fields below.